### ISSUE TITLE
jupyter_server's disable_check_xsrf ignored

### ISSUE THREAD

**Issue by cccs-nik** [jupyterhub/jupyterhub#4752]
### Bug description

As noted down in #4744, there are still issues with "HTTP 403: Forbidden ('_xsrf' argument missing from GET)" errors. More specifically, I'm having issues with [nbgallery](https://github.com/nbgallery/lab-extensions/blob/main/jupyterlab_nbgallery/handlers.py#L15-L34) and [jupyterlab_templates](https://github.com/finos/jupyterlab_templates/blob/main/jupyterlab_templates/extension.py#L99-L107).

Some sanitized logs which show the issue:
```
/opt/conda/lib/python3.10/site-packages/jupyter_server/serverapp.py:2235: JupyterServerAuthWarning: Core endpoints without @allow_unauthenticated, @ws_authenticated, nor @web.authenticated:
- GET of JupyterHubLogoutHandler registered for /user/$JUPYTERHUB_USER/logout
- GET of JupyterHubOAuthCallbackHandler registered for /user/$JUPYTERHUB_USER/oauth_callback
  self.web_app = ServerWebApplication(

[...]

[I 2024-03-25 18:27:05.673 ServerApp] Logged-in user {'kind': 'user', 'admin': True, 'name': '$JUPYTERHUB_USER', 'groups': [], 'session_id': 'some-session-id', 'scopes': ['access:servers!server=$JUPYTERHUB_USER/', 'read:users:groups!user=$JUPYTERHUB_USER', 'read:users:name!user=$JUPYTERHUB_USER']}
[E 2024-03-25 18:27:05.674 ServerApp] Not accepting cookie auth on GET /user/$JUPYTERHUB_USER/oauth_callback: HTTP 403: Forbidden ('_xsrf' argument missing from GET)
[I 2024-03-25 18:27:05.674 ServerApp] 302 GET /user/$JUPYTERHUB_USER/oauth_callback?code=[secret]&state=[secret] -> /user/$JUPYTERHUB_USER/templates/names (@127.0.0.1) 57.94ms

[...]

[W 2024-03-25 18:27:08.925 ServerApp] 403 GET /user/$JUPYTERHUB_USER/jupyterlab_nbgallery/environment?_=1711391214409 (@127.0.0.1) 2.76ms
[E 2024-03-25 18:27:08.925 ServerApp] Not accepting cookie auth on GET /user/$JUPYTERHUB_USER/jupyterlab_nbgallery/instrumentation: HTTP 403: Forbidden ('_xsrf' argument missing from GET)

[...]

[E 2024-03-25 18:48:44.067 ServerApp] Not accepting cookie auth on POST /user/$JUPYTERHUB_USER/post/example.ipynb: HTTP 403: Forbidden ('_xsrf' argument missing from POST)
[W 2024-03-25 18:48:44.067 ServerApp] 403 POST /user/$JUPYTERHUB_USER/post/example.ipynb (127.0.0.1): '_xsrf' argument missing from POST
[W 2024-03-25 18:48:44.068 ServerApp] wrote error: "'_xsrf' argument missing from POST"
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.10/site-packages/tornado/web.py", line 1769, in _execute
        result = await result  # type: ignore
      File "/opt/conda/lib/python3.10/site-packages/jupyter_server/base/handlers.py", line 750, in prepare
        await super().prepare()
      File "/opt/conda/lib/python3.10/site-packages/jupyter_server/base/handlers.py", line 632, in prepare
        self.check_xsrf_cookie()
      File "/opt/conda/lib/python3.10/site-packages/jupyterhub/services/auth.py", line 915, in <lambda>
        cls.check_xsrf_cookie = lambda handler: self.check_xsrf_cookie(handler)
      File "/opt/conda/lib/python3.10/site-packages/jupyterhub/services/auth.py", line 924, in check_xsrf_cookie
        check_xsrf_cookie(handler)
      File "/opt/conda/lib/python3.10/site-packages/jupyterhub/_xsrf_utils.py", line 181, in check_xsrf_cookie
        raise web.HTTPError(
    tornado.web.HTTPError: HTTP 403: Forbidden ('_xsrf' argument missing from POST)
[W 2024-03-25 18:48:44.069 ServerApp] 403 POST /user/$JUPYTERHUB_USER/post/example.ipynb (@127.0.0.1) 2.48ms
```
Seems like both extensions probably depended on "ServerApp.disable_check_xsrf=True" which is now being ignored.

### How to reproduce

Launch a JupyterHub server with a JupyterLab image that has jupyterlab_templates installed. The expected "template" button will be missing from the launcher and the logs will show 403s on jupyterlab_templates' handler.

#### Expected behaviour

"ServerApp.disable_check_xsrf=True" to be respected to allow extensions that depend on it to work.

#### Actual behaviour

See above.

### Your personal set up
Custom Helm chart built on top of zero-to-jupyterhub v3.3.3. 

 - OS:
 
 - Version(s): JupyterHub 4.1.1 and Python 3.10.
 

<details><summary>Full environment</summary>

```
# paste output of `pip freeze` or `conda list` here
```
</details>

<details><summary>Configuration</summary>

```python
# jupyterhub_config.py
```
</details>

<details><summary>Logs</summary>

```
# paste relevant logs here, if any
```
</details>

**Comment by minrk** [comment]
Fixed in 4.1.3

**Comment by cccs-nik** [comment]
Thanks for the quick update and for raising the detailed issue for nbgallery. The update definitely fixed the issue in our environment but I would be more than happy to not rely on disabling xsrf checks 🥳
